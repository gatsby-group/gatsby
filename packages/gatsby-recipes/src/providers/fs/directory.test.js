const directory = require(`./directory`)
const resourceTestHelper = require(`../resource-test-helper`)

const root = __dirname
// const content = `Hello, world!`
// const url = `https://gist.githubusercontent.com/KyleAMathews/3d763491e5c4c6396e1a6a626b2793ce/raw/545120bfecbe7b0f97f6f021801bc8b6370b5b41/gistfile1.txt`
// const url2 = `https://gist.githubusercontent.com/KyleAMathews/3d763491e5c4c6396e1a6a626b2793ce/raw/545120bfecbe7b0f97f6f021801bc8b6370b5b41/gistfile2.txt`

describe(`directory resource`, () => {
  test(`e2e directory resource test`, async () => {
    await resourceTestHelper({
      resourceModule: directory,
      resourceName: `Directory`,
      context: { root },
      initialObject: { path: `file.txt`, content },
      // partialUpdate: { content: content + `1` },
    })
  })
  // test(`e2e remote directory resource test`, async () => {
  //   await resourceTestHelper({
  //     resourceModule: file,
  //     resourceName: `Directory`,
  //     context: { root },
  //     initialObject: { path: `file.txt`, content: url },
  //     partialUpdate: { content: url2 },
  //   })
  // })
})
