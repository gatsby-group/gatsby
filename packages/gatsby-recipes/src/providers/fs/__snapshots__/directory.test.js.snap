// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directory resource e2e directory resource test: Directory create 1`] = `
Object {
  "_message": "Created directory src/image",
  "id": "src/image",
  "path": "src/image",
}
`;

exports[`directory resource e2e directory resource test: Directory create plan 1`] = `
Object {
  "describe": "Create src/image",
  "newState": "src/image",
}
`;

exports[`directory resource e2e directory resource test: Directory destroy 1`] = `
Object {
  "_message": "Created directory src/image",
  "id": "src/image",
  "path": "src/image",
}
`;

exports[`directory resource e2e directory resource test: Directory update 1`] = `
Object {
  "_message": "Created directory src/image",
  "id": "src/image",
  "path": "src/image",
}
`;

exports[`directory resource e2e directory resource test: Directory update plan 1`] = `
Object {
  "describe": "Create src/images",
  "newState": "src/images",
}
`;
